template/istqb.cls

```diff
@@ -106,44 +107,25 @@
         \g_istqb_schema_tl
     }
     { \\ \leavevmode \kern 0.1in }
+  % Level
   \tl_if_empty:NF
     \g_istqb_level_tl
-    {
-      \mbox { \g_istqb_level_tl }
-      \tl_if_empty:NF
-        \g_istqb_type_tl
-        { ~ }
-    }
-  \tl_if_empty:NF
-    \g_istqb_type_tl
-    { \mbox { \g_istqb_type_tl } }
-  \bool_if:nF
-    {
-      (
-        \tl_if_empty_p:N
-          \g_istqb_level_tl &&
-        \tl_if_empty_p:N
-          \g_istqb_type_tl
-      ) ||
-      (
-        \tl_if_empty_p:N
-          \g_istqb_title_tl &&
-        \tl_if_empty_p:N
-          \g_istqb_code_tl
-      )
-    }
-    { \,--\, }
+    { \mbox { \g_istqb_level_tl } }
+  % Title (code)
   \tl_if_empty:NF
     \g_istqb_title_tl
     {
-      \mbox { \g_istqb_title_tl }
+      \tl_if_empty:NF \g_istqb_level_tl { ~ }
+        {}~\mbox { \g_istqb_title_tl }
```

Comment:
Here, you should remove the `~` before `\mbox` to prevent a double `~~`. The `~` is now added on the previous line only if it is needed.

```suggestion
      \tl_if_empty:NF \g_istqb_level_tl { ~ }
      \mbox { \g_istqb_title_tl }
```